corehq/apps/userreports/reports/view.py

```diff
@@ -97,6 +98,20 @@ def query_dict_to_dict(query_dict, domain):
     return request_dict
 
 
+def _has_location_filter(view_fn, *args, **kwargs):
+    """check that the report has at least one location choice provider filter
+    """
+    # God help me.
+    # We can't access the instance of the report view in the location_safe
+    # decorator
```

Comment:
I don't think that would give access to the instance of the view class though, so I'd just need to use the same logic there. Unless I've missed something in what you were saying?